src/vm/jitinterface.cpp

```diff
@@ -6631,6 +6632,12 @@ const char* CEEInfo::getMethodNameFromMetadata(CORINFO_METHOD_HANDLE ftnHnd, con
         {
             MethodTable* pMT = ftn->GetMethodTable();
             classResult = pMT->GetFullyQualifiedNameInfo(&namespaceResult);
+            // Query enclosingClassName when it is required
+            // The namespace of nested classes is empty
+            if (namespaceResult[0] == '\0' && enclosingClassName != NULL)
+            {
+                pMT->GetMDImport()->GetNameOfTypeDef(pMT->GetEnclosingCl(), &enclosingResult, NULL);
```

Comment:
Don't you need to get the actual namespace here?

Reply:
No, we do not need namespace for nested classes.

Reply:
@jkotas  Oh, yes, we need the namespace (not just an empty string). Can we get it from other APIs?

Reply:
You will get it from the `GetNameOfTypeDef(pMT->GetEnclosingCl()`

Here is how it works:

Enclosing type: namespace="System.Runtime.Intrinsics.X86", name="Sse2"
Type: namespace="", name="X64"

Reply:
Thanks, will fix.